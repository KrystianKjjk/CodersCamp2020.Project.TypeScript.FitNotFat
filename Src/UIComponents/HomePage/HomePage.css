.main-text {
    font-weight: 400;
    font-size: 50px;
    letter-spacing: 0.055em;
    padding-bottom: 5%;
}

.features-text{
    font-size: 35px;
    line-height: 50px;
    letter-spacing: 0.05em;
    color:rgba(74, 72, 78, 0.9);
   padding-bottom: 5%;
}
.container-button {
    width: 70%;
    padding-top: 20px;
    display: flex;
    flex-direction: row;
    justify-content:space-between;
}

.container-button .sign-up-button,
.container-button .log-in-button{
    width: 200px;
}