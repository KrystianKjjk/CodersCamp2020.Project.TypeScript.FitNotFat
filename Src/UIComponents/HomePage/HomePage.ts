import {createElement} from '../../../Src/UIComponents/utils/utils';
import {generateRedButton,generateWhiteButton} from '../../../Src/UIComponents/Buttons/Buttons';

export function generateHomePageContent(onSignUpClick,onLogInClick):HTMLDivElement{
const homePageContentDiv=createElement('div','home-page-content-div')as HTMLDivElement;
const mainText=createElement('p','main-text','A little progress each day<br> adds up to big results')as HTMLParagraphElement;
const paragraph=createElement('p','features-text','Track calories<br>Break down ingredients<br>Log activities')as HTMLParagraphElement;

const signUpbutton = generateWhiteButton('Sign up', () => {
    onSignUpClick();
});
  const logInButton = generateRedButton('Log in', () => {
    onLogInClick();
  });
  
  signUpbutton.classList.add('sign-up-button');
  logInButton.classList.add('log-in-button');
  const containerButton = document.createElement('div');
  containerButton.classList.add('container-button');
  containerButton.append(signUpbutton, logInButton);

  homePageContentDiv.append(mainText,paragraph,containerButton);
  return homePageContentDiv;
}
