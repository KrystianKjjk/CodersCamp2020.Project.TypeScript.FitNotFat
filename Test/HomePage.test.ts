import { generateHomePageContent } from '../Src/UIComponents/HomePage/HomePage';

describe('home page tests', () => {
  let onSignUpClick;
  let onLogInClick;

  beforeEach(() => {
    onSignUpClick = jest.fn();
    onLogInClick=jest.fn();
  });

  test('should render home page component', () => {
    const homePage = generateHomePageContent(onSignUpClick, onLogInClick);
    expect(homePage).toMatchSnapshot();
  });

  test('should call onSignUpClick function after sign up button clicked', () => {
    const homePage = generateHomePageContent(onSignUpClick, onLogInClick);
    const signUpButton = homePage.querySelector(
      '.sign-up-button',
    ) as HTMLButtonElement;
    signUpButton.click();
    expect(onSignUpClick).toHaveBeenCalledTimes(1);
  });

  test('should call onLogInClick function after log in button clicked', () => {
    const homePage = generateHomePageContent(onSignUpClick, onLogInClick);
    const logInButton = homePage.querySelector(
      '.log-in-button',
    ) as HTMLButtonElement;
    logInButton.click();
    expect(onLogInClick).toHaveBeenCalledTimes(1);
  });
});

