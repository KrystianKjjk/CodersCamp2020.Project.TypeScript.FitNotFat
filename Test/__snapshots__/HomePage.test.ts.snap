// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home page tests should render home page component 1`] = `
<div
  class="home-page-content-div"
>
  <p
    class="main-text"
  >
    A little progress each day
    <br />
     adds up to big results
  </p>
  <p
    class="features-text"
  >
    Track calories
    <br />
    Break down ingredients
    <br />
    Log activities
  </p>
  <div
    class="container-button"
  >
    <button
      class="button sign-up-button"
      style="background-color: white; color: rgb(218, 27, 54); width: 200px; height: 50px;"
      type="button"
    />
    <button
      class="button log-in-button"
      style="background-color: rgb(218, 27, 54); color: white; width: 200px; height: 50px;"
      type="button"
    />
  </div>
</div>
`;
